### ISSUE TITLE
Definition file should contain an info object!

### ISSUE THREAD

**Issue by JesseObrien** [Surnet/swagger-jsdoc#71]
Trying to run:

`swagger-jsdoc.js -d swaggerdoc.js -o apidoc.json`

and getting this error:

`Definition file should contain an info object!
More at http://swagger.io/specification/#infoObject`

Here's my `swaggerdoc.js` definition file, not sure what's going on, I copied it from the examples I've seen here.

```var swaggerJSDoc = require('swagger-jsdoc');

var options = {
  swaggerDefinition: {
    swagger: '2.0',
    info: {
      title: 'Animal Tracker', // Title (required)
      version: '1.0.0', // Version (required)
      description: '',
      contact: {
        name: '',
        url: ''
      }
    },
  },
  apis: ['./app/**/*.ts'], // Path to the API docs
};

// Initialize swagger-jsdoc -> returns validated swagger spec in json format
var swaggerSpec = swaggerJSDoc(options);```

**Comment by kalinchernev** [comment]
Hi @JesseObrien
I think you've taken an example from code that is supposed to work without the CLI.
[That is the example definition file](https://github.com/Surnet/swagger-jsdoc/blob/master/example/swaggerDef.js) for usage in the CLI.
An example usage without the CLI would be [this](https://github.com/Surnet/swagger-jsdoc/blob/master/example/app.js#L36). The structure of the configurations are a bit different.
Can you try with these?

**Comment by JesseObrien** [comment]
Will try, thanks!

**Comment by JesseObrien** [comment]
Progress, or at least a different error. Now it outputs:

`You must provide arguments for reading APIs.`

I added the `apis` key to the definition file like so:

```module.exports = {
  info: { // API informations (required)
    title: 'Hello World', // Title (required)
    version: '1.0.0', // Version (required)
    description: 'A sample API', // Description (optional)
  },
  apis: ['./app/**/*.ts']
};```

**Comment by kalinchernev** [comment]
@JesseObrien I think that's good, as when using [the CLI it needs the files/folders](https://github.com/Surnet/swagger-jsdoc/blob/master/docs/CLI.md) with API documentation to be selected. Output spec file and watch flag are optional. The CLI works is an abstraction on top of the function that the module exports. So, you'll need to either use the module in code (as you do) and return the swagger spec on an endpoint as in the [getting started example](https://github.com/Surnet/swagger-jsdoc/blob/master/docs/GETTING-STARTED.md) as you started with the example, or delegate the parsing part to the CLI.

When I work with the module, I take the example definition (required arg) and just document my existing already working code (the input files with documentation also being required) and then I just run the module from the CLI. I don't require the module inside my app to parse and output the spec to an endpoint.

So basically there are 2 ways to parse comments with specification to an end specification, you can choose either way.

**Comment by JesseObrien** [comment]
hey @kalinchernev, thanks. I took a look at the code and made a quick change to allow passing in API paths in the definition file. Can you check the commit below to see if that's something we could pull in? Basically it looks to see if there's any arguments passed; if not AND there IS an api definition then use the paths in there.

https://github.com/JesseObrien/swagger-jsdoc/commit/0ed4ffe6b331114055019780c7f5e6afb769b620

**Comment by kalinchernev** [comment]
@JesseObrien thanks for investigating this and reaching this improvement - a pull would be more than welcome!
Could you please have a look at [this test](https://github.com/Surnet/swagger-jsdoc/blob/master/test/cli-test.js#L44) and maybe add a similar simple scenario for your case? That would solidify the cli with your scenario being handled.
Your suggestion is really appreciated!

**Comment by JesseObrien** [comment]
Absolutely, thanks!

**Comment by JesseObrien** [comment]
Alright, PR is up here. If the one test is not enough, I can add more 👍 
https://github.com/Surnet/swagger-jsdoc/pull/72

**Comment by kalinchernev** [comment]
Thanks @JesseObrien the pull request is fine I think, only minor things commented.

**Comment by JesseObrien** [comment]
Fixed and added this functionality with https://github.com/Surnet/swagger-jsdoc/pull/72
